healthcheck/audit.go

```diff
@@ -0,0 +1,54 @@
+package healthcheck
+
+import (
+	"context"
+	"time"
+
+	"github.com/cloudtrust/common-service/v2/events"
+	log "github.com/cloudtrust/common-service/v2/log"
+)
+
+type auditEventsReporterChecker struct {
+	alias    string
+	reporter events.AuditEventsReporterModule
+	timeout  time.Duration
+	response HealthStatus
+	logger   log.Logger
+}
+
+func newAuditEventsReporterChecker(alias string, reporter events.AuditEventsReporterModule, timeout time.Duration, cacheDuration time.Duration, logger log.Logger) BasicChecker {
+	healthStatusType := "auditEventreporter"
+	return &auditEventsReporterChecker{
+		alias:    alias,
+		reporter: reporter,
+		timeout:  timeout,
+		response: HealthStatus{Name: &alias, Type: &healthStatusType, CacheDuration: cacheDuration},
+		logger:   logger,
+	}
+}
+
+func (a *auditEventsReporterChecker) CheckStatus() HealthStatus {
+	if !a.response.hasExpired() {
+		return a.response
+	}
+
+	finished := make(chan bool)
+	go func() {
+		event := events.NewEvent("healthcheck", "LIVENESS_PROBE", "master", "health_checker", "health_checker", "master", map[string]string{})
+		a.reporter.ReportEvent(context.Background(), event)
+		finished <- true
+	}()
+
+	select {
```

Comment:
Wouldn't it be better to launch a refresh of the status without waiting for a response ? In one hand, if the connection is too long to establish, OKD could accumulate calls to the API and attempt multiple connections synchronously... In the other hand, if the timeout is too short, we might attempt/abort to many connections.